Refactor of the recipe editing page

rehandalal opened the pull request:
There a bunch of little usability tweaks in here but the bulk of this was to facilitate converting the recipe form to having it's own context so fields can be split out from the form component more easily and there's less need for deep prop drilling.

r?

rehandalal committed:
Refactor of the recipe editing page

tiftran commented on extension/content/components/pages/RecipeFormPage.js:
```diff
@@ -0,0 +1,53 @@
+          setImportInstructions(comment);
+        });
+    } else {
+      setData({});
```
was fiddling with it and the create workflow wasnt working for me until I inserted filter_object and arguments here...
not exactly sure if this is the right place to do that though...

tiftran reviewed (commented):
Thank you for making this some much more clean 🙏 

I tried experimenter -> ndt, experimenter sends action name which I guess since ndt is using action_id at the moment it didn't translate over?

rehandalal committed:
Merge branch 'master' into refactor-editor

rehandalal committed:
Address feedback

rehandalal commented:
@tiftran addressed the issues you mentioned and also made a small change to the name of the context since I think we will also use this context for the recipe details page.

tiftran commented:
bors r+
